docs/community/glossary.md

```diff
@@ -0,0 +1,15 @@
+| hook specification | A function signature that defines the API that a plugin developer must adhere to when writing the function that we promise to call somewhere in the napari codebase.                                                                                                                                                                        |
+| keybinding         | The key or key combination that causes some functions to occur.                                                                                                                                                                                                                                                                             |
+| labels             | The labels layer is an image where pixel values are integers. Each unique integer represents a unique object, feature or classification.                                                                                                                                                                                                    |
+| layer              | Layers are the viewable objects that can be added to napari viewer. Currently napari supports the following layers: image, labels, points, shapes, surface, tracks, vectors.                                                                                                                                                                |
+| mousebinding       | The specific mouse click that causes some functions to occur.                                                                                                                                                                                                                                                                               |
+| plugin             | A napari plugin is a specialized Python package which extends napari's functionality through additional file i/o support, image analysis, or other utilities. Plugin developers specify this functionality through a combination of Python entry points and a specialized manifest file. Plugins for napari can be found at napari-hub.org. |
+| shapes             | The shapes layer allows you to display a list of NxD arrays, where each array corresponds to one shape, specified by N points in D coordinates. Currently supported shapes: rectangle, ellipses, polygons, paths, and lines. Can be used as a visualization tool to highlight ROI, or be used for downstream operations.                    |
+| surface            | A surface comprises a set of triangles (typically in three dimensions) that are connected by their common edges or corners to form the mesh.                                                                                                                                                                                                |
```

Comment:
Is "vertices and edges" more accurate than "triangles"? 
I'm pretty sure in practice, triangles always get used, but you could certainly have surface meshes made out of rhombuses if you wanted.

Reply:
I think this definition is a bit technical... although maybe that's just because I don't really get it? In particular what makes a surface different to a shape? Maybe, what's it used for too?

Reply:
My understanding of Surface implementation in napari is mostly from the existing documentation: "The data for a surface layer is defined by a 3-tuple of its vertices, faces, and vertex values. The vertices are an NxD array of N vertices in D coordinates. The faces are an Mx3 integer array of the indices of the triangles making up the faces of the surface." Does that mean currently it's only possible to make triangles?

Some possible use cases: 3D area calculation, curvature measurements, surface mapping visualization, and in some other software Surface is almost equivalent to labels layer where it is used to define the boundary of objects.  
Because of its broad possibility (but currently very few implementations in napari), perhaps we leave it technical for now, until we have stronger use cases arise in the future?